sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SharedPreferencesManagerAndroidTest.java

```diff
@@ -102,7 +106,16 @@ private static SharedPreferencesTestData[] generateSharedPreferenceData() throws
 
         return testData;
     }
-
+    @Test
+    public void sharedPreferencesProtectedStorage() {
+        Boolean isProtectedStorage = context.isDeviceProtectedStorage();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```

Comment:
Condition in test function? Wouldn't it better to separate the testing on two test methods?